File: Homeworks 02/CS_Random_And_Timer/CS_Random_And_Timer/Form1.cs
namespace CS_Random_And_Timer
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        public Random randomNumber = new Random();
        public int occurencies1 = 0;
        public int occurencies2 = 0;
        public int occurencies3 = 0;
        public int occurencies4 = 0;
        public int occurencies5 = 0;
        public int occurencies6 = 0;
        public int totalOccurencies = 0;

        private void button1_Click(object sender, EventArgs e)
        {
            this.timer1.Interval = 1000;
            this.timer1.Start();
            this.button2.Visible = false;
        }

        private void button3_Click(object sender, EventArgs e)
        {
            this.timer1.Stop();
        }

        private void button2_Click(object sender, EventArgs e)
        {
            this.timer1.Interval = 10;
            this.timer1.Start();
            this.button1.Visible = false;
        }

        private void timer1_Tick(object sender, EventArgs e)
        {
            if (this.timer1.Interval == 10 && totalOccurencies > 999)
            {
                this.timer1.Stop();
            } else
            {
                int RandomInRange = randomNumber.Next(1, 7);
                this.richTextBox2.AppendText(RandomInRange + Environment.NewLine);

                totalOccurencies += 1;
                this.richTextBox9.Text = totalOccurencies.ToString();

                switch (RandomInRange)
                {
                    case 1:
                        occurencies1 += 1;
                        this.richTextBox3.Text = occurencies1.ToString();
                        break;
                    case 2:
                        occurencies2 += 1;
                        this.richTextBox4.Text = occurencies2.ToString();
                        break;
                    case 3:
                        occurencies3 += 1;
                        this.richTextBox5.Text = occurencies3.ToString();
                        break;
                    case 4:
                        occurencies4 += 1;
                        this.richTextBox6.Text = occurencies4.ToString();
                        break;
                    case 5:
                        occurencies5 += 1;
                        this.richTextBox7.Text = occurencies5.ToString();
                        break;
                    case 6:
                        occurencies6 += 1;
                        this.richTextBox8.Text = occurencies6.ToString();
                        break;
                    default:
                        this.timer1.Stop();
                        this.richTextBox1.Text = "Something went wrong!";
                        break;
                }
            }
        }
    }
}

File: Homeworks 02/VB_Random_And_Timer/VB_Random_And_Timer/Form1.vb
Imports System.Diagnostics.Eventing.Reader

Public Class Form1

    Public RandomNumber As New Random
    Public Occurencies1 As Integer = 0
    Public Occurencies2 As Integer = 0
    Public Occurencies3 As Integer = 0
    Public Occurencies4 As Integer = 0
    Public Occurencies5 As Integer = 0
    Public Occurencies6 As Integer = 0
    Public TotalOccurencies As Integer = 0

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Me.Timer1.Interval = 1000
        Me.Timer1.Start()
        Me.Button3.Visible = False
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Me.Timer1.Stop()
    End Sub

    Private Sub Button3_Click(sender As Object, e As EventArgs) Handles Button3.Click
        Me.Timer1.Interval = 10
        Me.Timer1.Start()
        Me.Button1.Visible = False
    End Sub

    Private Sub Timer1_Tick(sender As Object, e As EventArgs) Handles Timer1.Tick
        If Me.Timer1.Interval = 10 And TotalOccurencies > 999 Then
            Me.Timer1.Stop()
        Else
            Dim RandomInRange As Integer = RandomNumber.Next(1, 7)
            Me.RichTextBox2.AppendText(RandomInRange & Environment.NewLine)

            TotalOccurencies += 1
            Me.RichTextBox9.Text = TotalOccurencies

            Select Case RandomInRange
                Case 1
                    Occurencies1 += 1
                    Me.RichTextBox3.Text = Occurencies1
                Case 2
                    Occurencies2 += 1
                    Me.RichTextBox4.Text = Occurencies2
                Case 3
                    Occurencies3 += 1
                    Me.RichTextBox5.Text = Occurencies3
                Case 4
                    Occurencies4 += 1
                    Me.RichTextBox6.Text = Occurencies4
                Case 5
                    Occurencies5 += 1
                    Me.RichTextBox7.Text = Occurencies5
                Case 6
                    Occurencies6 += 1
                    Me.RichTextBox8.Text = Occurencies6
                Case Else
                    Me.Timer1.Stop()
                    Me.RichTextBox1.Text = "Something went wrong!"
            End Select
        End If
    End Sub
End Class
